src/studio/src/designer/backend/Controllers/DatamodelsController.cs

```diff
@@ -82,6 +84,12 @@ public async Task<IActionResult> UpdateDatamodel(string org, string app, string
                 MemoryStream jsonstream = new MemoryStream(byteArray);
                 await _repository.WriteData(org, app, $"{filePath}.schema.json", jsonstream);
 
+                // update meta data
+                JsonSchemaToInstanceModelGenerator converter = new JsonSchemaToInstanceModelGenerator(org, app, jsonSchemas);
+                ModelMetadata modelMetadata = converter.GetModelMetadata();
+                string root = modelMetadata.Elements != null && modelMetadata.Elements.Count > 0 ? modelMetadata.Elements.Values.First(e => e.ParentElement == null).TypeName : null;
+                _repository.UpdateApplicationWithAppLogicModel(org, app, modelName, "Altinn.App.Models." + root);
```

Comment:
I'm not quite sure about this change here.. Shouldn't it be possible to work with a data model without linking it to a specific application? Also, given that we are able to support multiple models in a single application a change like this would always replace the data model defined for task _1 with the model you are currently working on.

Reply:
My bad.. might be reading the code incorrectly.. It's added without a task ID. So only the point regarding this working without an application in the repository I'm worried about then.. 

up until this point I guess the repository could contain whatever and all datamodels will be stored in App/models..  But we don't require any of the other app structures.

Reply:
yes it won't work without an app, but you cannot get to the datamodelling page without first creating an app right now anyways. We will create support for shared models etc down the line, so it will probably be changed then